cachix/src/Cachix/Client/Config.hs

```diff
@@ -35,24 +36,23 @@ mkConfig authtoken = Config
   , binaryCaches = []
   }
 
-readConfig :: IO (Maybe Config)
-readConfig = do
-  filename <- getFilename
+type ConfigPath = FilePath
+
+readConfig :: ConfigPath -> IO (Maybe Config)
+readConfig filename = do
   doesExist <- doesFileExist filename
   if doesExist
   then Just <$> input auto (toS filename)
   else return Nothing
 
-
-getFilename :: IO FilePath
-getFilename = do
+getDefaultFilename :: IO FilePath
+getDefaultFilename = do
   dir <- getXdgDirectory XdgConfig "cachix"
   createDirectoryIfMissing True dir
```

Comment:
Side effect in something IO action called `get...`.

Reply:
Just a line below there is `getXdgDirectory` - what's wrong with this naming?

Reply:
Ah! That's a valid point.